.github/workflows/scanner-nvd-update.yaml

```
@@ -77,12 +77,11 @@ jobs:
         path: nvd-api.zip
         if-no-files-found: error
 
-  upload:
+  upload-nvd-feeds:
     needs:
     - fetch-nvd-feeds
-    - fetch-nvd-api
     runs-on: ubuntu-latest
-    if: github.event_name != 'pull_request'
+    # if: github.event_name != 'pull_request'
```

I was expecting a single change like:

```suggestion
    # if: github.event_name != 'pull_request' && (needs.fetch-nvd-feeds.result == 'success' || needs.fetch-nvd-api.result == 'success')
```

Maybe I missed something, can you clarify? Thanks.

---

I split the upload step.
If we do 
```
needs:
 - fetch-nvd-feeds
 - fetch-nvd-api
```
The upload step is going to be dependent on both to be successful regardless `if`
cc: @jvdm

---

> The upload step is going to be dependent on both to be successful regardless if

I don't think that's entirely correct, see [by default if uses `success()`](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#status-check-functions), but if you use a condition this will overwritten and the step will be executed if any of the above is true.

---

I looked closer into this, I think this explains the behavior: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#failure-with-conditions

> You can include extra conditions for a step to run after a failure, but you must still include failure() to override the default status check of success() that is automatically applied to if conditions that don't contain a status check function.

I think it means GH adds a `success()` by default in the `if`. Then we would have to overwrite that explicitly by adding `failure()`, or `always()`...

So we would have to have to use something like:

```
if: ${{ (success() || failure()) && github.event_name != 'pull_request' }}
```

Which is confusing, and I think adding the two steps is just easier for us to avoid this (un)pleasant ambiguity.